src/test/test-compress.c

```diff
@@ -29,6 +30,15 @@
 # define LZ4_OK -EPROTONOSUPPORT
 #endif
 
+#if HAVE_LZ4
+static void *lz4_dl = NULL;
+
+static DLSYM_FUNCTION(LZ4_compress_default);
+static DLSYM_FUNCTION(LZ4_decompress_safe);
+static DLSYM_FUNCTION(LZ4_decompress_safe_partial);
+static DLSYM_FUNCTION(LZ4_versionNumber);
+#endif
```

Comment:
This seems a bit messy. What about moving these to compress.h using `DLSYM_PROTOTYPE` and making `dlopen_lz4()` a public function? Then the existing `dlopen_lz4()` function can be reused instead of declaring a new one here.